docs/sdk-and-tools/mxpy/smart-contract-interactions.md

```diff
@@ -122,55 +140,60 @@ myNonPayableEndpoint() {
+Besides the function and arguments parts, the snippet is more or less the same as when deploying or upgrading a contract. When calling a non payable function, we need to provide the endpoint's name as the function argument. As for the arguments, they have to be in the **same order** as in the smart contract, including when calling an endpoint that has a variable number of arguments. Now, for the sake of example, we provided the arguments in multiple ways.
+
+It is up to each developer to choose the layout he prefers, but a few points need to be underlined:
+
+- Most of the supplied **arguments** need to be in the **hexadecimal format**: `0x...`.
+- When converting a value to a hexadecimal format, we need to make sure it has an **even number** of characters. If not, we need to provide an extra `0` in order to make it even:
+  - Example: the number `911` -> in hexadecimal encoding, it is equal to: `38f` -> so we need to provide the argument `0x038f`.
 - Arguments can be provided both as a fixed arguments (usually for unchangeable arguments like the contract's address or a fixed number) or can be provided as an input in the terminal, when interacting with the snippet (mostly used for arguments that change often like numbers).
 
-In our example we provide the address argument as a fixed argument. We then convert it to hex format (as it is in the bech32 format by default) and only after that we pass it as a parameter. As for the BigUint parameters, we provide the first two parameters directly in the terminal and the last one as a fixed argument, hex encoded.
+In our example we provide the address argument as a fixed argument. We then convert it to hexadecimal format (as it is in the bech32 format by default) and only after that we pass it as a parameter. As for the `BigUint` parameters, we provide the first two parameters directly in the terminal and the last one as a fixed argument, hexadecimal encoded.
 
 :::tip
-mxpy facilitates us with some encoding conventions, including:
+`mxpy` facilitates us with some encoding conventions, including:
 
-- We can use **str:** for encoding strings. For example: str:MYTOKEN-123456
-- Blockchain addresses that start with **erd1** are automatically encoded, so there is no need to further hex encode them
-- The values **true** or **false** are automatically converted to **boolean** values
-- Values that are identified as **numbers** are hex encoded by default
-- Arguments like **0x...** are left unchanged, as they are interpreted as already encoded hex values
+- We can use `str:` for encoding strings. For example: `str:MYTOKEN-123456`.
+- Blockchain addresses that start with `erd1` are automatically encoded, so there is no need to further hex encode them.
+- The values **true** or **false** are automatically converted to **boolean** values.
+- Values that are identified as **numbers** are hex encoded by default.
```

Comment:
numbers are hex encoded as `BigUInt` values. Maybe that could be wrong in some cases.
e.g. the contract expects a UInt8Value.